l10n_nl_tax_declaration_reporting/report/report_vat.py

```diff
@@ -0,0 +1,161 @@
+        _4BO = self._get_line_value('4b', 'omzet')
+        _4BB = self._get_line_value('4b', 'btw')
+        _5BB = self._get_line_value('5b', 'btw')
+
+        _5AB = _1AB + _1BB + _1CB + _1DB + _2AB - _4AB - _4BB
+        _5CB = _5AB - _5BB
+
+        docargs = {
+            'time_now': fields.Datetime.now(),
+            'company': company_name,
+            'start_date': date_from,
+            'end_date': date_to,
+            '_1AO': _1AO,
+            '_1AB': _1AB,
+            '_1BO': _1BO,
+            '_1BB': _1BB,
+            '_1CO': _1CO,
+            '_1CB': _1CB,
+            '_1DO': _1DO,
+            '_1DB': _1DB,
+            '_1EO': _1EO,
+            '_2AO': _2AO,
+            '_2AB': _2AB,
+            '_3AO': _3AO,
+            '_3BO': _3BO,
+            '_3CO': _3CO,
+            '_4AO': _4AO,
+            '_4AB': _4AB,
+            '_4BO': _4BO,
+            '_4BB': _4BB,
+            '_5AB': _5AB,
+            '_5BB': _5BB,
+            '_5CB': _5CB,
+            'tax_code_error': self.tax_code_error,
+        }
+
+        return self.env['report'].render(
+            'l10n_nl_tax_declaration_reporting.tax_report_template', docargs)
+
+    def _get_line_value(self, code, type):
+        tax_report_lines = self.tax_report_lines
+        map_tax_tag = self.map_tax_tag
+        tag_id = self.tags_map[code][type]
+        for line in tax_report_lines:
+            tax_id = line[0]
+            tags_list = map_tax_tag[tax_id]
+            if tag_id in tags_list:
+                if type == 'omzet':
+                    return line[2]
+                return line[1]
+        return 0.0
+
+    def _map_tax_tag(self, form):
+
+        query = """
+            SELECT account_tax_id, account_account_tag_id
+            FROM account_tax_account_tag
+        """
+        self._cr.execute(query)
+
+        res = self._cr.fetchall()
+        map = {}
+        for row in res:
+            key = row[0]
+            if row[0] not in map:
+                map[key] = [row[1]]
+            else:
+                map[key].append(row[1])
+        return map
+
+    def _get_lines(self, form):
+
+        def filter_by_dates(date_from, date_to):
+            where = 'WHERE inv.state IN (\'open\',\'paid\') '
+            if date_from:
+                where += " and inv.date >=  '" + date_from + "'"
+            if date_to:
+                where += " and inv.date <=  '" + date_to + "'"
+            return where
+
+        query = """
+            SELECT inv_tax.tax_id,
+                   sum(inv_tax.amount),
+                   sum(inv_line.price_subtotal_signed)
+            FROM account_invoice_tax AS inv_tax
+                JOIN account_invoice_line_tax inv_line_tax
+                    on inv_line_tax.tax_id = inv_tax.tax_id
+                JOIN account_invoice_line inv_line
+                    on inv_line.id = inv_line_tax.invoice_line_id
+                JOIN account_invoice inv
+                    on inv.id = inv_line.invoice_id
```

Comment:
in my version for 9, I changed the query entirely, because the way the query stands here, it will return the amounts multiplied by the amount of lines which have the same tax (note that inv_tax.amount is the sum of *all* lines with the tax in the invoice, not just the amount of some specific line): https://github.com/OCA/l10n-netherlands/pull/68#issuecomment-285956281